### ISSUE TITLE
Refactor accessibility guidelines to remove single window assumption

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#119364]
The `MinimumTextContrastGuideline` matcher assumes that all widgets are painted into the same view:

https://github.com/flutter/flutter/blob/b2f2bf31c0fe83be2e6ed208d71c6c808993e3c5/packages/flutter_test/lib/src/accessibility.dart#L404

In a multi-window world this is not true. Instead, the matcher needs to look up the View of a given element via the `View.of(context)` API (or alternatively walk the widget tree up until it reaches a view widget).

**Comment by goderbauer** [comment]
Basically, we have to replace the `tester.binding.window` int the code snippet above with the actual `FlutterView` the node in question is being rendered into. We can obtain that node by finding the `View` ancestor and grabbing its `view` property, which is a FlutterView.

**Comment by goderbauer** [comment]
cc @chunhtai @hangyujin Maybe one of you would be interested in taking a taking a look at this?

**Comment by chunhtai** [comment]
tentatively assign to myself.

**Comment by goderbauer** [comment]
This problem is not limited to the `MinimumTextContrastGuideline `, some of the other guidelines in `packages/flutter_test/lib/src/accessibility.dart` also have this problem.

**Comment by pdblasi-google** [comment]
I _think_ this could be handled with `tester.viewOf` now. Pass it a `find.byElementPredicate((Element e) => e == element)` (it's a bit roundabout, but it should work without an API change) and you should be able to get the related view now and after multi-view is fully introduced.

I can look into this one since I've been working in the area if @chunhtai is okay with passing it off?

**Comment by chunhtai** [comment]
@pdblasi-google unassinged myself, I have not started yet
